Open store for writing review on iOS

soerenBoisen opened the pull request:
- add: on iOS calling openStoreScreen does not open the write review form as the documentation states that it should; according to Apples documentation a query param must be added to actually open the review

See: https://developer.apple.com/documentation/storekit/requesting_app_store_reviews?language=objc

Closes #5

soerenBoisen committed:
Open store for writing review on iOS

- add: on iOS calling openStoreScreen does not open the write review form as
the documentation states that it should; according to Apples documentation a
query param must be added to actually open the review

See: https://developer.apple.com/documentation/storekit/requesting_app_store_reviews?language=objc

soerenBoisen commented:
@chemerisuk Can you have a quick look at this? It is a pretty small change.

chemerisuk commented:
@soerenBoisen the logic is that `openStoreScreen` method is used to open app page in app/play stores. `requestReview` method is used to open dialog that allows to send review.

I don't like your change because it changes that. Why can't you use `requestReview`?

soerenBoisen commented:
@chemerisuk Your documentation says:

> Launches App/Play store page with a review form.

So I assumed that it was supposed to open the store page and show review form. Your Android screenshots seem to indicate that as well? (Didn't try it on Android yet, so don't know how it actually works).

This is what the change does. Without it, `openStoreScreen` will simply open App Store for the app and the user has to scroll down and find the review action.

The reason I want to use `openStoreScreen` in my app, is to provide a permanent shortcut to review the app from the in-app settings. I cannot use `requestReview` for that, since it may or may not open a review dialog depending on, how many times it has been shown to the user already.

Since this action is user-initiated and not a spontaneous popup, it should not be subject to rate-limiting.

I am fine with adding a separate JS function to do it, if you want to preserve current functionality?

Something like `openStoreForReview` maybe?

chemerisuk commented on www/AppReview.js:
```diff
@@ -35,6 +35,7 @@ exports.openStoreScreen =
  */
 function(packageName) {
```
Let's move `writeReview` to be function argument and be falsy by default. This flag will work only for iOS
```suggestion
function(packageName, writeReview) {
```
Also don't forget to update jsdocs comment with param `writeReview` description.

chemerisuk commented on src/ios/AppReviewPlugin.m:
```diff
@@ -20,12 +20,19 @@ - (void)openStoreScreen:(CDVInvokedUrlCommand*)command {
     if (trackId) {
         NSString* storeURL = [NSString stringWithFormat:@"https://apps.apple.com/app/id%@", trackId];
-        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:storeURL] options:@{} completionHandler:nil];
+        NSString* storeReviewURL = [NSString stringWithFormat:@"https://apps.apple.com/app/id%@?action=write-review", trackId];
```
DRY code
```suggestion
        if (writeReview) {
          storeURL = [NSString stringWithFormat:@"%@?action=write-review", storeURL];
        }
```
So you don't need if statement at line 31

chemerisuk reviewed (changes requested):
@soerenBoisen Let's add new flag to `openStoreScreen` in order to implement what you want. See my comments below.

soerenBoisen committed:
Apply review comments

- add: make opening review form an option in the JS API
- cleanup: simplify Obj-C code

soerenBoisen commented:
@chemerisuk Applied suggested changes. Is the JSDoc okay?

chemerisuk commented:
@soerenBoisen looks good, thanks!
